app/ts/background/windows/confirmTransaction.ts

```diff
@@ -25,25 +25,36 @@ let openedDialog: PopupOrTab | undefined = undefined
 let pendingTransactions = new Map<string, Future<Confirmation>>()
 const pendingConfirmationSemaphore = new Semaphore(1)
 
-export async function updateConfirmTransactionViewWithPendingTransaction() {
+export async function updateConfirmTransactionViewWithPendingTransaction(ethereumClientService: EthereumClientService) {
 	const promises = await getPendingTransactions()
 	if (promises.length > 0) {
-		await sendPopupMessageToOpenWindows({ method: 'popup_update_confirm_transaction_dialog', data: promises })
+		const currentBlockNumberPromise = ethereumClientService.getBlockNumber()
+		await sendPopupMessageToOpenWindows({ method: 'popup_update_confirm_transaction_dialog', data: { pendingTransactions: promises, currentBlockNumber: await currentBlockNumberPromise }})
 		return true
 	}
 	return false
 }
 
-export async function updateConfirmTransactionViewWithPendingTransactionOrClose() {
-	if (await updateConfirmTransactionViewWithPendingTransaction() === true) return
-	if (openedDialog) await closePopupOrTabById(openedDialog.popupOrTab)
+export async function updateConfirmTransactionViewWithPendingTransactionOrClose(ethereumClientService: EthereumClientService) {
+	if (await updateConfirmTransactionViewWithPendingTransaction(ethereumClientService) === true) return
+	if (openedDialog) await closePopupOrTabById(openedDialog)
 	openedDialog = undefined
 }
 
+export const isConfirmTransactionFocused = async () => {
+	if (openedDialog === undefined) return false
+	const pendingTransactions = await getPendingTransactions()
+	if (pendingTransactions[0] === undefined) return false
+	const popup = await getPopupOrTabOnlyById(pendingTransactions[0].popupOrTabId)
```

Comment:
Convenience alternative

```suggestion
	const [pendingTransaction] = await getPendingTransactions()
	if (pendingTransaction === undefined) return false
	const popup = await getPopupOrTabOnlyById(pendingTransaction.popupOrTabId)
```